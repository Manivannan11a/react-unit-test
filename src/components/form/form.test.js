import React from 'react';
import { shallow } from 'enzyme';
import FormComp from '.';
import { fireEvent } from '@testing-library/react';


describe('Form Test', () => {
    let wrapper;
    // const setState = jest.fn();
    // const useStateSpy = jest.spyOn(React, "useState")
    // useStateSpy.mockImplementation((init) => [init, setState]);

    it('renders without crashing', () => {
        expect(shallow(<FormComp />)).toMatchSnapshot();
    });

    beforeEach(() => {
        wrapper = shallow(<FormComp />);
        // wrapper.toMatchSnapshot();
        // console.log(wrapper.state())
    });
    // afterEach(() => {
    //     jest.clearAllMocks();
    // });

    // eslint-disable-next-line jest/no-identical-title
    it("renders without crashing", () => {
        expect(shallow(<FormComp />)).toMatchSnapshot();
    });

    it("check field values", () => {
        const fname = wrapper.find("input[name='fname']");
        fname.value = "test";
        // fname.simulate('change', { target: { name: 'fname', value: "test" } });
        // fireEvent.change(fname);
        console.log(wrapper, "simulate");
        // wrapper.find('[id="submit"]').click  
        expect(fname.value).toEqual("test");
    })
});
