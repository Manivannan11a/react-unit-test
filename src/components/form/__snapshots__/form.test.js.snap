// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Test renders without crashing 1`] = `ShallowWrapper {}`;

exports[`Form Test renders without crashing 2`] = `ShallowWrapper {}`;
